File: .net & mono/Weed3Demo/config/cache/MemCache.cs
using System;
using System.Collections.Generic;
using System.Text;
using MemcachedClientLibrary;

using Noear.Weed.Cache;

namespace Weed3Demo.config 
{
    internal class MemCache : ICacheService
    {
        MemcachedClient _cache = null;

        public MemCache(string keyHead, int defSeconds, string host) {
            this._CacheKeyHead = keyHead;
            this._DefalutSeconds = defSeconds;

            InitMemCache(host);
        }

        private string _CacheKeyHead;
        public string getCacheKeyHead()
        {
            return _CacheKeyHead; 
        }

        private int _DefalutSeconds;
        public int getDefalutSeconds()
        {
            return _DefalutSeconds; 
        }

        public object get(string key)
        {
                return _cache.Get(key);
        }

        public void remove(string key)
        {
                _cache.Delete(key);
        }

        public void store(string key, object obj, int seconds)
        {
                _cache.Set(key, obj, DateTime.Now.AddSeconds(seconds));
        }


        void InitMemCache(string host) //ip:poart
        {
            //以下代码，要以具体使用的 Memcached Client 来处理
            //
            string[] servers = { host };

            //缓存服务设置，配置优先级高，无配置使用默认值

            if (!MemcachedClient.Exists(host)) {
                MemcachedClient.Setup(host, servers);
            }

            //实例化
            _cache = MemcachedClient.GetInstance(host);


            //检查缓存服务是否开启

            _cache.SendReceiveTimeout = 1000;
            _cache.MinPoolSize = 10;
            _cache.MaxPoolSize = 100;
        }
    }
}

File: java/Weed3Demo/src/weed3demo/config/cache/MemCache.java
package weed3demo.config.cache;

import net.spy.memcached.AddrUtil;
import net.spy.memcached.ConnectionFactoryBuilder;
import net.spy.memcached.MemcachedClient;
import net.spy.memcached.auth.AuthDescriptor;
import net.spy.memcached.auth.PlainCallbackHandler;
import noear.weed.cache.ICacheService;
import weed3demo.config.utils.EncryptUtil;

/**
 * Created by noear on 2017/4/25.
 */

public final class MemCache implements ICacheService {
    private String _cacheKeyHead;
    private int _defaultSeconds;

    private MemcachedClient mcc = null;

    public MemCache(String keyHeader, int defSeconds, String server, String user, String password) {
        _cacheKeyHead = keyHeader;
        _defaultSeconds = defSeconds;

        try {
            if (server.indexOf("aliyun") > 0) {
                //1.阿里云
                AuthDescriptor ad = new AuthDescriptor(new String[]{"PLAIN"},
                        new PlainCallbackHandler(user, password));

                mcc = new MemcachedClient(new ConnectionFactoryBuilder()
                        .setProtocol(ConnectionFactoryBuilder.Protocol.BINARY)
                        .setAuthDescriptor(ad).build(),
                        AddrUtil.getAddresses(server));
            } else {
                //2.本地
                mcc = new MemcachedClient(new ConnectionFactoryBuilder()
                        .setProtocol(ConnectionFactoryBuilder.Protocol.BINARY).build(),
                        AddrUtil.getAddresses(server));
            }
        } catch (Exception ex) {
            ex.printStackTrace();
        }
    }

    public void store(String key, Object obj, int seconds) {

        if (mcc != null) {
            String newKey = newKey(key);
            try {
                mcc.set(newKey, seconds, obj);
            }catch (Exception ex) {
                ex.printStackTrace();
            }
        }
    }

    public Object get(String key) {
        if (mcc != null ) {
            String newKey = newKey(key);
            try {
                return mcc.get(newKey);
            } catch (Exception ex) {
                ex.printStackTrace();
                return null;
            }
        } else
            return null;
    }

    public void remove(String key) {
        if (mcc != null) {
            String newKey = newKey(key);
            mcc.delete(newKey);
        }
    }

    public int getDefalutSeconds()
    {
        return _defaultSeconds;
    }

    public String getCacheKeyHead()
    {
        return _cacheKeyHead;
    }

    private String newKey(String key) {
        return _cacheKeyHead + "$" + EncryptUtil.md5(key);
    }
}